src/mat.rs

```diff
@@ -1356,6 +1356,40 @@ macro_rules! mat4s {
                 (*self * point.into_homogeneous_point()).normalized_homogeneous_point().truncated()
             }
 
+            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
+            /// `T` is a translation and `R` a rotation), return the `translation` field of the
+            /// isometry.
+            ///
+            /// If `self` does not represent an isometry, the returned value has undefined
+            /// properties.
+            #[inline]
+            pub fn extract_translation(&self) -> $v3t {
+                self.cols[3].truncated()
+            }
+
+            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
```

Comment:
Would be more appropriate to just say "if the upper left 3x3 matrix in this matrix is a pure rotation matrix" or something to that effect